app/scripts/metamask-controller.js

```diff
@@ -5261,6 +5272,10 @@ export default class MetamaskController extends EventEmitter {
 
     this.unMarkPasswordForgotten();
 
+    // If the wallet is locked when the `controllerConnectionChanged` event is
+    // emitted, we need to start polling for balances as soon as it is unlocked.
+    this.triggerNetworkrequests();
```

Comment:
I wonder how much time it takes to get here after the user clicks the button on the unlock screen, and whether it is worth triggering these network requests at the same time that the password is submitted.

Reply:
On my machine, it takes 156ms. It would be good to keep this time as low as possible. I am not sure if we want to trigger the requests if a user enters a wrong password, though, in which case we need to wait for the response from the submission outcome from the KeyringController.